clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -248,4 +261,28 @@ public void updateAssignment(ConsumerGroupHeartbeatResponseData.Assignment assig
         }
         maybeTransitionToStable();
     }
+
+    @Override
+    public void completeReconcile(Set<TopicPartition> revokedPartitions,
+                                  Set<TopicPartition> assignedPartitions,
+                                  Optional<KafkaException> callbackError) {
+        if (callbackError.isPresent()) {
+            // TODO: how to react to callback errors?
+        }
+
+        assignmentReconciler.completeReconcile(revokedPartitions, assignedPartitions);
+        transitionTo(MemberState.STABLE);
+        // TODO: update state to signal the HeartbeatRequestManager to send an ACK heartbeat
```

Comment:
We need to sort this out. Wondering if new states/transitions, used from the HB poll would do...as described in my other comment [here](https://github.com/apache/kafka/pull/14357/files#r1361441480)?